tools/transform.cpp

```diff
@@ -436,15 +436,15 @@ check_refinement(Errors &errs, Transform &t, State &src_state, State &tgt_state,
     return;
   }
 
-  auto mk_fml = [&](expr &&refines) -> expr {
+  auto mk_fml = [&](const expr &refines) -> expr {
```

Comment:
std::bind wasn't working well for lambda functions that have arguments with rvalue reference.
The reason is that std::bind internally forwards the argument to a variable and calls the function: https://stackoverflow.com/a/34878230